### DISCUSSION TITLE
[Bug]

### DISCUSSION THREAD

**Discussion by phpMuhammed** [Laravel-Backpack/community-forum#380] [Q&A (Help)] [answered]
How to print data from a list page after filtering it. The problem is that the query returns the same without filtering.
For your information, I use a special button and file for printing.

**Comment by tabacitu** [comment]
Hi Muhammed. 

I'm afraid I don't understand what you're trying to do and cannot achieve it. Can you please expand and/or provide screenshots?

Cheers!

**Reply by phpMuhammed** [reply]
Hello,
this->crud->query value after filtering the data by filters return same as data before filtration.
so, if I have an action like "print today appointments" it will print all data although it is supposed to bring the filtered data.

I hope I explained it in a clear way.

**Reply by pxpm** [reply, marked as answer]
@phpMuhammed have a look at the ListOperation search code, where we apply the filters/search to the query:
 https://github.com/Laravel-Backpack/CRUD/blob/98c72342e4c0b2ff0f56ec0376f9f819d77fd6f2/src/app/Http/Controllers/Operations/ListOperation.php#L70

Cheers

**Comment by o15a3d4l11s2** [comment]
@phpMuhammed, do you want to be able to print a full CRUD list (possibly filtered), or is it enough to have a "Print Today Appointments" button that prints today's appointments no matter what is show inside the CRUD list at the moment?

If the second option is enough, there is no need to do it through Backpack's list operation logic and you can just write your own select query that gets all the appointments for today and then print them. The button would just call your own function.

**Reply by phpMuhammed** [reply]
I want to print the filtered data only. the print will execute by custom action.

**Reply by o15a3d4l11s2** [reply]
Ok, so the button is more like "Print list". Then exactly as Pedro explained, you will have to implement a logic similar to the `search` and it will provide you the exact result set as in the list.

**Reply by phpMuhammed** [reply]
Thanks for you all,
I will try it.
